jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -49,4 +64,85 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     }
     return JarType.STANDARD;
   }
+
+  /**
+   * Explode jar and create three layers for classes, resources and dependencies on container.
+   *
+   * @param jarPath path to jar file.
+   * @return list of {@link FileEntriesLayer}.
+   * @throws IOException if I/O error occurs when opening the file.
+   */
+  public static List<FileEntriesLayer> explodeStandardJar(Path jarPath) throws IOException {
+    TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider();
+    Path tempDirectoryPath = tempDirectoryProvider.newDirectory();
+    ZipUtil.unzip(jarPath, tempDirectoryPath);
+
+    List<FileEntriesLayer> layers = new ArrayList<>();
+    Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
+    Predicate<Path> isResourceFile = path -> !path.getFileName().toString().endsWith(".class");
+
+    // Determine class and resource files in the directory containing jar contents and create
+    // FileEntriesLayer for each type of layer (class or resource), while maintaining the
+    // file's original project structure.
+    FileEntriesLayer classesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                tempDirectoryPath,
+                isClassFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("Classes")
+            .build();
+    FileEntriesLayer resourcesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                tempDirectoryPath,
+                isResourceFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("Resources")
+            .build();
+
+    // Get dependencies from Class-Path in the jar's manifest and create a FileEntriesLayer.Builder
+    // with these dependencies as entries.
+    List<Path> dependencies = new ArrayList<>();
+    JarFile jarFile = new JarFile(jarPath.toFile());
+    String classPath = jarFile.getManifest().getMainAttributes().getValue("Class-Path");
+    if (classPath != null) {
+      for (String dependency : Splitter.on(" ").split(classPath)) {
+        Path depPath = Paths.get(dependency);
+        dependencies.add(depPath);
```

Comment:
super nit:

```suggestion
        dependencies.add(Paths.get(dependency));
```

Just FYI, another way is
```
.splitToList().stream().map(Paths::get).collect(Collectors.toList());
```

Reply:
Pretty neat.